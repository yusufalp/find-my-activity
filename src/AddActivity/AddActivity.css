.required {
  margin: 0 auto;
  text-align: right;
  padding-top: 24px;
}

.error {
  margin: auto;
  color: #C58882;
  margin-top: 12px
}