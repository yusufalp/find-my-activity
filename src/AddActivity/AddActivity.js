import React from 'react';
import ActivityContext from '../ActivityContext/ActivityContext';
import './AddActivity.css'

class AddActivity extends React.Component {
  static contextType = ActivityContext;
  state = {
    categories: this.context.categories,
    activities: this.context.activities,
    name: '',
    content: '',
    duration: '',
    materials: '',
  }
  handleDataSubmit(e) {
    e.preventDefault();
    const name = e.target.name.value
    const content = e.target.content.value
    const duration = e.target.duration.value
    const materials = e.target.materials.value
    const ageGroup = e.target.ageGroup.value
    const category = e.target.categories.value
    const newActivity = {
      name,
      content,
      duration,
      materials,
      ageGroup,
      category
    }
    this.context.updateData(newActivity);
    this.props.history.push('/');
  }
  handleNameChange(value) {
    this.setState({
      name: value.trim()
    })
  }
  handleContentChange(value) {
    this.setState({
      content: value.trim()
    })
  }
  handleDurationChange(value) {
    this.setState({
      duration: value.trim()
    })
  }
  handleMaterialsChange(value) {
    this.setState({
      materials: value.trim()
    })
  }
  render() {
    return (
      <div>
        <header>
          <h2>Add Activity</h2>
          <p>
            In this section, you can share your activity with our community by filling out the 
            form below and let people find it by searching through our list of activities. 
            Thank you in advance for sharing.
          </p>
        </header>
        <section>
          <form method="get" onSubmit={e => this.handleDataSubmit(e)}>
            <label htmlFor="name">Name</label>
            <input type="text" name="name" id="name" onChange={e => this.handleNameChange(e.target.value)} />
            <label htmlFor="content">Content</label>
            <textarea type="text" name="content" id="content" rows="6" onChange={e => this.handleContentChange(e.target.value)} />
            <label htmlFor="duration">Duration</label>
            <input type="text" name="duration" id="duration" onChange={e => this.handleDurationChange(e.target.value)} />
            <label htmlFor="materials">Materials</label>
            <input type="text" name="materials" id="materials" onChange={e => this.handleMaterialsChange(e.target.value)} />
            <label htmlFor="ageGroup">Age Group</label>
            <select name="ageGroup" id="ageGroup" required>
              <option value="">Select an age group</option>
              <option>Option 1</option>
              <option>Option 2</option>
              <option>Option 3</option>
              <option>Option 4</option>
            </select>
            <label htmlFor="categories">Categories</label>
            <select name="categories" id="categories" required>
              <option value="">Select a category</option>
              {this.state.categories.map((category, i) =>
                <option key={i}>{category.category}</option>
              )}
            </select>
            <button className='add-button' type='submit'>Submit</button>
          </form>
        </section>
      </div>
    )
  }
}

export default AddActivity;