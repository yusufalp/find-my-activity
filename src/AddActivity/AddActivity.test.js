import React from 'react';
import ReactDOM from 'react-dom';
import AddActivity from './AddActivity';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<AddActivity />, div);
  ReactDOM.unmountComponentAtNode(div);
});
