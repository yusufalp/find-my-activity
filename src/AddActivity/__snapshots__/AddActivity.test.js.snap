// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddActivity component renders the UI as expected 1`] = `
<div>
  <header>
    <h2>
      Add Activity
    </h2>
    <p>
      In this section, you can share your activity with our community by filling out the form below and let people find it by searching through our list of activities. Thank you in advance for sharing.
    </p>
  </header>
  <section>
    <form
      method="get"
      onSubmit={[Function]}
    >
      <label
        htmlFor="name"
      >
        Name*
      </label>
      <input
        id="name"
        name="name"
        onChange={[Function]}
        placeholder="Drawing"
        required={true}
        type="text"
      />
      <label
        htmlFor="content"
      >
        Content*
      </label>
      <textarea
        id="content"
        name="content"
        onChange={[Function]}
        placeholder="Explain in details..."
        required={true}
        rows="6"
        type="text"
      />
      <label
        htmlFor="duration"
      >
        Duration* 
        <span>
          (in minutes, numbers only)
        </span>
      </label>
      <input
        id="duration"
        min="0"
        name="duration"
        onChange={[Function]}
        placeholder="45"
        required={true}
        type="number"
      />
      <label
        htmlFor="materials"
      >
        Materials
      </label>
      <input
        id="materials"
        name="materials"
        onChange={[Function]}
        placeholder="Pencils, markers and paper"
        type="text"
      />
      <label
        htmlFor="agegroup"
      >
        Age Group*
      </label>
      <select
        id="agegroup"
        name="agegroup"
        required={true}
      >
        <option
          value=""
        >
          Select an age group
        </option>
        <option
          value="1"
        >
          1
        </option>
        <option
          value="2"
        >
          2
        </option>
        <option
          value="3"
        >
          3
        </option>
        <option
          value="4"
        >
          4
        </option>
        <option
          value="5"
        >
          5
        </option>
        <option
          value="6"
        >
          6
        </option>
        <option
          value="7"
        >
          7
        </option>
        <option
          value="8"
        >
          8
        </option>
      </select>
      <label
        htmlFor="categories"
      >
        Categories*
      </label>
      <select
        id="categories"
        name="categories"
        required={true}
      >
        <option
          value=""
        >
          Select a category
        </option>
      </select>
      <p
        className="required"
      >
        * = required fields
      </p>
      <p
        className="error"
      />
      <button
        className="add-button"
        type="submit"
      >
        Submit
      </button>
    </form>
  </section>
  <img
    alt="children with activities"
    className="group-kids"
    src="/images/activities.png"
  />
</div>
`;
